### DISCUSSION TITLE
github actions / ppooll-machine automated build frequency

### DISCUSSION THREAD

**Discussion by chausch** [ppooll-dev/ppooll#123] [General]
hi guys,
the auto-versioning system is really cool, however the amount of releases is a bit much.
it's cool that the version number increases with every commit now, but i think we don't have to make a release out of every commit. what do you think @jpsteccato @klausfilip ?
maybe it would be possible to only release every few hours or once per day? or maybe even only once a week?
or would that introduce incompatibilities with the upgrade check? (i.e. checking but finding no release?)

**Comment by jpsteccato** [comment]
i definitely agree, i will change the Github Action to do a "nightly" build if anything on main has changed.

what ive been wondering with out current workflow is how do we increase the semantic version numbers? (ie 8.6.0 -> 8.6.1)

i have an idea for a workflow that might work for us all:
- let's say we are currently at vers 8.6.0.  development of new features/major changes should happen on a branch called "dev-8.7.0"
- when we encounter a bug, we fix the bug on the main branch, which creates a release and updates ppooll vers to 8.6.1
- then merge changes (manually if same patch modified) from the main branch with the bug fix to "dev-8.7.0"
- when "dev-8.7.0" is ready for release, merge into main to create new release

**Reply by chausch** [reply]
that sounds good, but requires a lot of structure and discipline from us 😅

**Reply by jpsteccato** [reply]
it's definitely more to be aware of when making changes (whether it's feature or bug fix) -- but i dont think it will be _much_ more.

we would work out dev-8.7.0 for 90% of the time.  if a bug comes up that should be fixed immediately (or we need to test something on what would be the current, "stable" version, we can just switch the branch, make the change, switch back and merge from the changes.

**Comment by jpsteccato** [comment]
only building nightly releases has been implemented!  but now we should probably get rid of the "release count" number at the end of the version (so its just 8.6.0.231018).  i can make this change next

**Comment by chausch** [comment]
> only building nightly releases has been implemented! but now we should probably get rid of the "release count" number at the end of the version (so its just 8.6.0.231018). i can make this change next

i'm open to however you would like to do this!

**Comment by chausch** [comment]
ah, a small posscriptum: we could still be creating multiple releases per day, manually, so we could still keep the count …

**Reply by jpsteccato** [reply]
oh interesting, i dont see us having to do this too often, but I will add the "count" back.  

but we should really avoid changing the version numbers ourselves via texteditor-- should i create a separate action for creating a new release "on-demand"?  so if you want to make a new release, you have to: 
1) push changes to main 
2) go to Github Actions and manually run "create-release" action.

let me know if this makes sense and i can set it up

**Reply by jpsteccato** [reply]
went ahead and did this -- you can create a release manually by going to Actions -> Jobs (left side) -> Manual Release
